### PR TITLE
Fix: Getting the size from github rest api

### LINKED ISSUE #31: Display the Git-Repository size instead of total file size (or both?)
**Is your feature request related to a problem? Please describe.**
It would be nice to know the total size of a Git repository before starting to clone. However, currently, the extension only displays the total size as considered by all files currently in the selected branch (afaik). This means that it is useful for knowing the approximate size of `Download code as zip`, but not for the git repo itself.

**Describe the solution you'd like**
I'd want to see the entire size instead.

**Describe alternatives you've considered**
Looking at the API manually is not nice.

**Additional context**
Current calculation: <https://github.com/AminoffZ/github-repo-size/blob/main/src/scripts/internal/dom-manipulation.ts#L154-L156>
Implementation idea: <https://stackoverflow.com/questions/8646517/how-can-i-see-the-size-of-a-github-repository-before-cloning-it>
=> using the GitHub API, get infos about the entire repository

### PR DESCRIPTION
**Pull request by mAmineChniti**
Closes #31 
- Getting repo size directly using github rest api
- Formatting the size from KB to the proper format accordingly

### CODE REVIEW

**Comment by AminoffZ** [comment]
Theoritically this makes sense, however, the size returned by that API call is inaccurate. I'm getting 1.19MB as total size. The current implementation with this repo as test subject at least results in an accurate decompressed main branch download size. Neither implementation at the moments take into account other branches.

To replicate:
- Download the source code
- Extract the zip
- See folder size

**Comment by mAmineChniti** [comment]
@AminoffZ i think the truth lies somewhere in the middle the api call gives me 2.19 MB, the file size counter gives 1.31 MB, same time downloading and decompressing gives 1.30 MB and cloning gives 3.3MB, so i find it weird that one set of files on one specific branch that hasn't changed has so many changing size values based on the download/calculation method, what should we do?

**Comment by XZTDean** [comment]
I think the size return from API is counting all the storage space taken on github. It is not only counting the default branch file sizes. 
I have a repo that the default branch only contains a readme file, but the size from API is much larger than the readme file size. The API value is almost equal to the .git folder size.

**Review by AminoffZ — CHANGES_REQUESTED** [review]
TLDR: __User Value__ from current implementation > suggested change.

1. Until we can reliably fetch / calculate the clone size, we should show source code folder size (current implementation).
2. When we have clone size, we could show both (UI suggestions welcome)
